core/controllers/base.py

```diff
@@ -218,12 +216,14 @@ def dispatch(self):
             return
 
         if self.user_is_scheduled_for_deletion:
+            # In DEV_MODE, clearing cookies does not log out the user, so we
+            # force-clear them by redirecting to the logout URL.
```

Comment:
FYI, though the comment is useful, it doesn't address what I was asking. I was wondering why we don't redirect them to create_logout_url() on prod -- it seems safer and simpler to just do that in all cases?

Reply:
I modified the code and discovered that we still have this issue #7836 :sweat_smile:, I will fix this tommorow in this or separate PR.